cleanlab/experimental/datalab/datalab.py

```diff
@@ -0,0 +1,385 @@
+        """
+        if issue_types is not None:
+            issue_types_copy = issue_types.copy()
+            self._check_missing_args(required_defaults_dict, issue_types_copy)
+        else:
+            issue_types_copy = required_defaults_dict.copy()
+        # Check that all required arguments are provided.
+        self._validate_issue_types_dict(issue_types_copy, required_defaults_dict)
+
+        # Remove None values from argument list, rely on default values in IssueManager
+        for key, value in issue_types_copy.items():
+            issue_types_copy[key] = {k: v for k, v in value.items() if v is not None}
+        return issue_types_copy
+
+    @staticmethod
+    def _check_missing_args(required_defaults_dict, issue_types):
+        for key, issue_type_value in issue_types.items():
+            missing_args = set(required_defaults_dict.get(key, {})) - set(issue_type_value.keys())
+            # Impute missing arguments with default values.
+            missing_dict = {
+                missing_arg: required_defaults_dict[key][missing_arg]
+                for missing_arg in missing_args
+            }
+            issue_types[key].update(missing_dict)
+
+    @staticmethod
+    def _validate_issue_types_dict(
+        issue_types: Dict[str, Any], required_defaults_dict: Dict[str, Any]
+    ) -> None:
+        missing_required_args_dict = {}
+        for issue_name, required_args in required_defaults_dict.items():
+            if issue_name in issue_types:
+                missing_args = set(required_args.keys()) - set(issue_types[issue_name].keys())
+                if missing_args:
+                    missing_required_args_dict[issue_name] = missing_args
+        if any(missing_required_args_dict.values()):
+            error_message = ""
+            for issue_name, missing_required_args in missing_required_args_dict.items():
+                error_message += f"Required argument {missing_required_args} for issue type {issue_name} was not provided.\n"
+            raise ValueError(error_message)
+
+    def _get_report(self, k: int, verbosity: int) -> str:
+        # Sort issues based on the score
+        # Show top K issues
+        # Show the info (get_info) with some verbosity level
+        #   E.g. for label issues, only show the confident joint computed with the health_summary
+        issue_type_sorted_keys: List[str] = (
+            self.issue_summary.sort_values(by="score", ascending=True)["issue_type"]
+            .to_numpy()
+            .tolist()
+        )
+        issue_managers = self._get_managers(issue_type_sorted_keys)
+        report_str = ""
+        for issue_manager in issue_managers:
+            report_str += issue_manager.report(k=k, verbosity=verbosity) + "\n\n"
+        return report_str
+
+    def _get_managers(self, keys: List[str]) -> List[IssueManager]:
+        issue_managers = [self.issue_managers[i] for i in keys]
+        return issue_managers
+
+    def find_issues(
```

Comment:
Is it intended behaviour that calling `lab.find_issues()` with no args passed in will just do nothing? 

You've probably thought this through more re the implementation, but from a user's POV it could be confusing that the method just does nothing silently, I think it might be nice to give some sort of indication (print statement?) that datalab is not doing the checks bc of the lack of info

Reply:
Atm. we assume `lab.find_issues` is provided with some arguments.

There are some print statements that track the progress now. 

I'll remember to 

- [ ] Print warning message when no issue check is performed in a call to `Datalab.find_issues`.